packages/contracts/contracts/ReputationProofVerifier.sol

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {IUnirep} from './interfaces/IUnirep.sol';
+import {IVerifier} from './interfaces/IVerifier.sol';
+
+import {EpochKeyProofVerifier} from './EpochKeyProofVerifier.sol';
+
+contract ReputationProofVerifier is IUnirep {
+    IVerifier public reputationVerifier;
+    EpochKeyProofVerifier public epochKeyProofVerifier;
+
+    uint256 public constant SNARK_SCALAR_FIELD =
+        21888242871839275222246405745257275088548364400416034343698204186575808495617;
+
+    constructor(
+        IVerifier _reputationVerifier,
+        EpochKeyProofVerifier _epochKeyProofVerifier
+    ) {
+        reputationVerifier = _reputationVerifier;
+        epochKeyProofVerifier = _epochKeyProofVerifier;
+    }
+
+    function decodeReputationSignals(
+        uint256[] calldata publicSignals
+    ) public view returns (ReputationSignals memory) {
+        ReputationSignals memory signals;
+        signals.epochKey = publicSignals[0];
+        signals.stateTreeRoot = publicSignals[1];
+        signals.graffitiPreImage = publicSignals[4];
+        // now decode the control values
+        (
+            signals.revealNonce,
+            signals.attesterId,
+            signals.epoch,
+            signals.nonce
+        ) = epochKeyProofVerifier.decodeEpochKeyControl(publicSignals[2]);
+
+        (
+            signals.minRep,
+            signals.maxRep,
+            signals.proveMinRep,
+            signals.proveMaxRep,
+            signals.proveZeroRep,
+            signals.proveGraffiti
+        ) = decodeReputationControl(publicSignals[3]);
+        return signals;
+    }
+
+    function decodeReputationControl(
+        uint256 control
+    )
+        public
+        pure
+        returns (
+            uint256 minRep,
+            uint256 maxRep,
+            uint256 proveMinRep,
+            uint256 proveMaxRep,
+            uint256 proveZeroRep,
+            uint256 proveGraffiti
+        )
+    {
+        minRep = control & ((1 << 64) - 1);
+        maxRep = (control >> 64) & ((1 << 64) - 1);
+        proveMinRep = (control >> 128) & 1;
+        proveMaxRep = (control >> 129) & 1;
+        proveZeroRep = (control >> 130) & 1;
+        proveGraffiti = (control >> 131) & 1;
+        return (
+            minRep,
+            maxRep,
+            proveMinRep,
+            proveMaxRep,
+            proveZeroRep,
+            proveGraffiti
+        );
+    }
+
+    function verifyReputationProof(
+        uint256[] calldata publicSignals,
+        uint256[8] calldata proof,
+        uint48 attesterCurrentEpoch,
+        bool attesterStateTreeRootExists
```

Comment:
Same here, should return a `ReputationSignals` object for further verification.

Reply:
Change added